assets/js/state/softwareUpdates.js

```diff
@@ -0,0 +1,56 @@
+import { createAction, createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  loading: false,
+  softwareUpdates: {},
+  errors: [],
+};
+
+export const softwareUpdatesSlice = createSlice({
+  name: 'softwareUpdates',
+  initialState,
+  reducers: {
+    startLoadingSoftwareUpdates: (state) => {
+      state.loading = true;
+    },
+    setSoftwareUpdates: (
+      state,
+      { payload: { hostId, relevant_patches, upgradable_packages } }
+    ) => {
+      state.loading = false;
+
+      state.softwareUpdates = {
+        ...state.softwareUpdates,
+        [hostId]: {
+          relevant_patches,
+          upgradable_packages,
+        },
+      };
+    },
+    setEmptySoftwareUpdates: (state, { payload: { hostId } }) => {
+      state.loading = false;
+      state.softwareUpdates = Object.entries(state.softwareUpdates).reduce(
+        (newState, [id, updates]) =>
+          id !== hostId ? { ...newState, [id]: updates } : newState,
+        {}
+      );
```

Comment:
```suggestion
     state.softwareUpdates = {
        ...state.softwareUpdates,
        [hostId]: {},
      };
```

What do you think?

Reply:
Sure, I think it's good. It got me thinking though; what do we think about clearing setting `[hostId]: { relevant_patches: [], upgradable_packages: [] }` as opposed to `[hostId]: {}`?

Reply:
On second thought, it might be bad as the selector would report `relevant_patches`/`upgradable_packages == 0` instead of `undefined`. But as @nelsonkopliku mentioned, error handling should handle this before it even hits the selector :thinking:

Reply:
No strong opinion, let's ship whatever you prefer :+1:

Reply:
I like having a semantic difference between `undefined` vs `0`. Sure, the error handling should handle this but it still gives a more semantically correct representation of the state of the system.